bftengine/include/bftengine/Replica.hpp

```diff
@@ -50,20 +55,28 @@ enum MsgFlag : uint8_t {
 // 2. Key exchange
 // 3. Change DB scheme
 // and basically any management action that is handled by the layer that uses concord-bft.
-class ControlHandlers {
+class IControlHandler {
  public:
+  static const std::shared_ptr<IControlHandler> instance(IControlHandler *ch = nullptr) {
```

Comment:
Why are we applying a singleton pattern here? Can't we instantiate one ControlHandler and pass around its reference?

Reply:
In the design we prefer to have a singleton on top of the complication of passing the reference throughout the whole stack